### PR TITLE
Added Dependency Category support

### PR DESCRIPTION
**Pull request by ankit-privado**
Added dependency category, process, rule in dependency sheet
Ticket: https://privado-ai.atlassian.net/browse/LE-274

### CODE REVIEW

File: src/main/scala/ai/privado/entrypoint/ScanProcessor.scala [hunk 1]
```diff
@@ -173,7 +176,37 @@ object ScanProcessor extends CommandProcessor {
                             language = Language.withNameWithDefault(pathTree.last)
                           )
                         )
-                        .filter(filterSystemConfigByLang)
+                        .filter(filterSystemConfigByLang),
+                      auditCollection = configAndRules.auditCollection
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:180]: What are these 3 new rules and will this be part of Privado repo?

Comment (ankit-privado) [reply src/main/scala/ai/privado/entrypoint/ScanProcessor.scala:180]: There three new rule is for audit dependency only. so to identify which dependency belong to which category (Collection, Web client, Utility) we need rules. Yes, we also need to define the rule inside Privado repo.

**Comment by khemrajrathore** [comment]
Changes good to Me, but will wait for the PR of Privado

**Review by pandurangpatil — APPROVED** [review]
looks good

**Pull request merged**